opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/AbstractDaoHibernate.java

```diff
@@ -259,6 +271,53 @@ public void deleteAll(final Collection<T> entities) throws DataAccessException {
 
     @Override
     public List<T> findMatching(final org.opennms.core.criteria.Criteria criteria) {
+        //Below findMultiAndMatching method supports multiAnd criteria, which is added to support
+        // multiple "and" condition on same columns of a table.
+        //e.g  select * from event_parameters where (eventparam1_.name= "instance" and eventparam1_.value = "node1")
+        // and (eventparam1_.name= "trigger" and eventparam1_.value = "3.0")
+        if(criteria.isMultipleAnd()){
+            return findMultiAndMatching(criteria);
+        } else {
+            final HibernateCallback<List<T>> callback = buildHibernateCallback(criteria);
+            return getHibernateTemplate().execute(callback);
+        }
+    }
+
+    private List<T> findMultiAndMatching(final org.opennms.core.criteria.Criteria criteria){
+        Set<T> allUniqueRecords = new LinkedHashSet<>();
+        Collection<Restriction> allRestrictions = criteria.getRestrictions();
+
+        //set of multiand restrictions
+        Set<Restriction> multiAndRestrictionSet = allRestrictions.stream().filter(
+                restriction -> restriction.getType().equals(Restriction.RestrictionType.MULTIAND)).collect(Collectors.toSet());
+
+        //set of non multiand restrictions
+        Set<Restriction>   nonMultiAndRestrictionSet =  Sets.difference(new HashSet<Restriction>(allRestrictions),multiAndRestrictionSet);
+
+        //iterating multiand and setting nonmultiand + single multiand restriction in criteria,
+        // retrieve result set, create intersection of each query executed
+        multiAndRestrictionSet.stream().forEach(restriction ->{
+            Collection<Restriction> allMultiAndRestrictions = ((AllRestriction) restriction).getRestrictions();
+            allMultiAndRestrictions.stream().forEach(singleMultiAndRestriction ->{
+                org.opennms.core.criteria.Criteria copyOfCriteria = criteria.clone();
+                copyOfCriteria.setRestrictions(nonMultiAndRestrictionSet);
+                copyOfCriteria.addRestriction(singleMultiAndRestriction);
+                try {
```

Comment:
try/catch should be moved to `getQueryResult`

Reply:
try catch moved to getQueryResult